contracts/base/MarketsState.sol

```diff
@@ -61,7 +70,12 @@ contract MarketsState is
         address collateralAsset,
         uint256 amount
     ) external onlyWhitelisted() isInitialized() {
-        markets[borrowedAsset][collateralAsset].increaseRepayment(amount);
+        address cTokenAddress = _getCTokenAddress(borrowedAsset);
+        if (cTokenAddress.isEmpty()) {
+            markets[borrowedAsset][collateralAsset].increaseRepayment(amount);
+        } else {
+            markets[cTokenAddress][collateralAsset].increaseRepayment(amount);
+        }
```

Comment:
Because the same logic is duplicated several times in this file, we should instead change the `_getCTokeAddress` function to `_getMarket` and just use that to perform the action needed in each function.